frontend/test/framework/testDependencies.cpp

```diff
@@ -394,16 +414,118 @@ static void test5() {
   assert(nQueryFourRuns == 1);
   assert(nQueryFiveRuns == 2);
   assert(nQuerySixRuns == 3);
+  assert(nQuerySevenRuns == 0);
   assert(q6 == "hello there giant worldhello there giant world");
 }
 
+// Pre-test for test6b
+// query three's result doesn't change even when query one's result changes, so
+// query seven is not recomputed.
+static void test6a() {
+  Context ctx;
+  Context* context = &ctx;
+
+  // get a baseline
+  printf("part 1\n");
+  inputString = "hello world";
+  nInputQueryRuns = 0;
+  nQueryOneRuns = 0;
+  nQueryTwoRuns = 0;
+  nQueryThreeRuns = 0;
+  nQueryFourRuns = 0;
+  nQueryFiveRuns = 0;
+  nQuerySixRuns = 0;
+  nQuerySevenRuns = 0;
+
+  std::ignore = querySeven(context, /* unused */ 0);
+  assert(nInputQueryRuns == 1);
+  assert(nQueryOneRuns == 1);
+  assert(nQueryTwoRuns == 0);
+  assert(nQueryThreeRuns == 1);
+  assert(nQueryFourRuns == 0);
+  assert(nQueryFiveRuns == 0);
+  assert(nQuerySixRuns == 0);
+  assert(nQuerySevenRuns == 1);
+
+  context->advanceToNextRevision(false);
+  printf("part 2\n");
+  inputString = "world hello"; /* Same parity as hello world, should not re-compute query 7. */
+  std::ignore = querySeven(context, /* unused */ 0);
+  assert(nInputQueryRuns == 2);
+  assert(nQueryOneRuns == 2);
+  assert(nQueryTwoRuns == 0);
+  assert(nQueryThreeRuns == 2);
+  assert(nQueryFourRuns == 0);
+  assert(nQueryFiveRuns == 0);
+  assert(nQuerySixRuns == 0);
+  assert(nQuerySevenRuns == 1);
+}
+
+// This is pretty much the same as 6a, except the sub-computation of 3->1->i
+// is computed separately before 7 is ever invoked. This reproduces a bug
+// in the query system in which query 7 could be made to depend on query 1,
+// thus making it be recomputed.
```

Comment:
Is this a current bug? I thought you had fixed it in this PR? Was I misunderstanding?

Reply:
The bug is fixed now, I believe, this is a regression test. The comment can be reworded to make it clear that the bug isn't present anymore.